pkg/traceql/engine.go

```diff
@@ -232,3 +343,48 @@ func (s Static) asAnyValue() *common_v1.AnyValue {
 		},
 	}
 }
+
+// Regex to extract matchers from a query string
+// This regular expression matches a string that contains three groups separated by operators.
+// The first group is a string of alphabetical characters, dots, and underscores.
+// The second group is a comparison operator, which can be one of several possibilities, including =, >, <, and !=.
+// The third group is one of several possible values: a string enclosed in double quotes,
+// a number with an optional time unit (such as "ns", "ms", "s", "m", or "h"),
+// a plain number, or the boolean values "true" or "false".
+// Example: "http.status_code = 200" from the query "{ .http.status_code = 200 && .http.method = }"
+var matchersRegexp = regexp.MustCompile(`[a-zA-Z._]+\s*[=|<=|>=|=~|!=|>|<|!~]\s*(?:"[a-zA-Z._]+"|[0-9smh]+|true|false)`)
```

Comment:
i'm concerned about these regex's being difficult to maintain as we grow this functionality. frankly they may already be difficult to maintain.

i think the right choice is likely an error tolerant traceql parser, but i'm fine with this approach for now.

Reply:
I share your concerns. I dislike the solution, but it's functional and I think it's the best we've got for now :(.